addons/website/models/website.py

```diff
@@ -443,10 +454,55 @@ def set_images(images):
         }
         custom_resources = self._website_api_rpc('/api/website/1/configurator/custom_resources/%s' % kwargs.get('industry_id'), params)
 
+        # Update CTA
+        cta_data = website.get_cta_data(kwargs.get('website_purpose'), kwargs.get('website_type'))
+        if cta_data['cta_btn_text']:
+            xpath_view = 'website.snippets'
+            parent_view = self.env['website'].with_context(website_id=website.id).viewref(xpath_view)
+            self.env['ir.ui.view'].create({
+                'name': parent_view.key + ' CTA',
+                'key': parent_view.key + "_cta",
+                'inherit_id': parent_view.id,
+                'website_id': website.id,
+                'type': 'qweb',
+                'priority': 32,
+                'arch_db': """
+                    <data>
+                        <xpath expr="//t[@t-set='cta_btn_href']" position="replace">
+                            <t t-set="cta_btn_href">%s</t>
+                        </xpath>
+                        <xpath expr="//t[@t-set='cta_btn_text']" position="replace">
+                            <t t-set="cta_btn_text">%s</t>
+                        </xpath>
+                    </data>
+                """ % (cta_data['cta_btn_href'], cta_data['cta_btn_text'])
+            })
+            header_ids = [
+                'website.template_header_default_oe_structure_header_default_1',
+                'website.template_header_hamburger_oe_structure_header_hamburger_1',
+                'website.template_header_slogan_oe_structure_header_slogan_2',
+                'website.template_header_boxed_oe_structure_header_boxed_2',
+                'website.template_header_image_oe_structure_header_image_2',
+            ]
+            for header_id in header_ids:
+                try:
+                    view_id = self.env['website'].viewref(header_id)
+                    if view_id:
+                        header_arch = pycompat.to_text(view_id.arch_db)
+                        el = html.fragment_fromstring(header_arch, create_parent=True)
```

Comment:
archdb and view in Odoo is xml based :/ don't try to parse it in html... or at least never dump it as html

```python
> arch = "<img src='ok'/>"     
                                                                                                                                                                         
> etree.tostring(etree.fromstring(arch))                                                                                                                                                                
b'<img src="ok"/>'

> html.tostring(html.fromstring(arch))                                                                                                                                                                  
b'<img src="ok">'
```

Reply:
btw, why use fragment and create_parent... since it is already a uniq element -> fromstring. And since we have a root element, no parent required